tf2_geometry_msgs/scripts/test.py

```diff
@@ -135,6 +135,42 @@ def test_transform(self):
         self.assertEqual(out.vector.y, 0)
         self.assertEqual(out.vector.z, 0)
 
+        # Testing for pose and covariance transform
+        t = TransformStamped()
+        t.transform.translation.x = 1.0
+        t.transform.translation.y = 2.0
+        t.transform.translation.z = 3.0
+        t.transform.rotation = Quaternion(w=0.0, x=1.0, y=0.0, z=0.0)
+
+        v = PoseWithCovarianceStamped()
+        v.header.stamp = rclpy.time.Time(seconds=2.0).to_msg()
+        v.header.frame_id = 'a'
+        v.pose.covariance = (
+          1.0, 2.0, 3.0, 4.0, 5.0, 6.0,
+          1.0, 2.0, 3.0, 4.0, 5.0, 6.0,
+          1.0, 2.0, 3.0, 4.0, 5.0, 6.0,
+          1.0, 2.0, 3.0, 4.0, 5.0, 6.0,
+          1.0, 2.0, 3.0, 4.0, 5.0, 6.0,
+          1.0, 2.0, 3.0, 4.0, 5.0, 6.0
+        )
+        v.pose.pose.position.x = 1.0
+        v.pose.pose.position.y = 2.0
+        v.pose.pose.position.z = 3.0
+        v.pose.pose.orientation = Quaternion(w=0.0, x=1.0, y=0.0, z=0.0)
+
+        out = tf2_geometry_msgs.do_transform_pose_with_covariance_stamped(v, t)
+        expected_covariance = (
+          1.0, 0.0, 0.0, 4.0, 0.0, 0.0,
+          0.0, 2.0, 0.0, 0.0, 5.0, 0.0,
+          0.0, 0.0, 3.0, 0.0, 0.0, 6.0,
+          1.0, 0.0, 0.0, 4.0, 0.0, 0.0,
+          0.0, 2.0, 0.0, 0.0, 5.0, 0.0,
+          0.0, 0.0, 3.0, 0.0, 0.0, 6.0
+        )
+        self.assertEqual(out.pose.pose.position.x, 2)
+        self.assertEqual(out.pose.pose.position.y, 0)
+        self.assertEqual(out.pose.pose.position.z, 0)
+        self.assertEqual(out.pose.covariance, expected_covariance)
```

Comment:
You should also check that the output orientations are correct.
```suggestion
        self.assertEqual(out.pose.pose.position.x, 2)
        self.assertEqual(out.pose.pose.position.y, 0)
        self.assertEqual(out.pose.pose.position.z, 0)
        self.assertEqual(out.pose.orientation.x, 0)
        self.assertEqual(out.pose.orientation.y, 0)
        self.assertEqual(out.pose.orientation.z, 0)
        self.assertEqual(out.pose.orientation.w, 1)
        self.assertEqual(out.pose.covariance, expected_covariance)
```

Reply:
For line 173 specifically: `out.pose.covariance` is an `np.array` with shape `(36,)`, while `expected_covariance` is a tuple. As such, their comparison won't really work. The comparison here will return an `np.array` with shape `(36,)` filled with boolean values corresponding to element-wise comparisons - so a quick hack could be to 
```py
        self.assertTrue(np.all(out.pose.covariance == expected_covariance))
```
Another solution could be to convert `expected_covariance` to a `numpy` type, and then use [`np.array_equiv`](https://numpy.org/doc/stable/reference/generated/numpy.array_equiv.html#numpy.array_equiv).

Reply:
When I try the above in terminal, it gives me the following:
```
>>> np.all(out.pose.covariance == expected_covariance)
<stdin>:1: DeprecationWarning: elementwise comparison failed; this will raise an error in the future.
False
```
So I will change the `expected_covariance` to a `np.array` of shape `(36,)`